timon/configure.py

```diff
@@ -228,6 +229,76 @@ def sort_key_func(kval):
     return ordered_cfg
 
 
+def get_data(fieldname, entry):
+    field = fieldname.split(".")
+    rslt = entry
+    for data in field:
+        if rslt[data]:
+            rslt = rslt[data]
+    return rslt
+
+
+def mk_nested_dict(grpby, entries):
+    rslt = {}
+    print("rslt = %8x" % id(rslt))
+    for key, entry in entries.items():
+        target = rslt
+        for grp_info in grpby[:-1]:
+            fieldname = grp_info["field"]
+            dflt_val = grp_info.get("default", grp_info["values"][0])
+            field = get_data(fieldname, entry)
+            if field not in target:
+                target[field] = {}
+                print("create {} %x for field %8s in %x" % (
+                      id(target[field]), field, id(target)))
+            target = target[field]
+
+        # Handle last group differenty
+        grp_info = grpby[-1]
+        fieldname = grp_info["field"]
+        dflt_val = grp_info.get("default", grp_info["values"][0])
+        field =  get_data(fieldname, entry)
+        if field not in target:
+            target[field] = []
+        target = target[field]
+        target.append(key)
+        target.sort()
+    return rslt
+
+
+def dicdic2lisdic(dicdic, grpby, lvl=0):
+    grp = grpby[lvl]
+    grpvals = grp["values"]
+    def keyfunc(key):
```

Comment:
check flake, but I think there should be an empty line before declaration of keyfunc